### ISSUE TITLE
Status of Bootstrap 4, AdminLTE 3

### ISSUE THREAD

**Issue by tacman** [kevinpapst/AdminLTEBundle#161]
Congrats on the release of AdminLTEBundle 3.3!

It appears that this bundle is based on Bootstrap 3 and the AdminLTE package version 2 (which uses version 3).

AdminLTE (from Colorlib) version 3 uses Bootstrap 4.

You have a branch call adminlte-3, which is NOT the same as the Version 3 of AdminLTE Bundle that you just released.  It hasn't been updated since January of 2020.

This has left me a bit confused.  Bootstrap 5 is coming very soon (the fine folks behind it have said they expect very few changes between beta 2, which is out now, and the final version).   I've almost given up on ColorLib ever releasing version 3 based on bootstrap 4, since it's been in beta since 2019.  However, the master branch is now 3.1-rc and uses bs4.

I had a fork based on the adminlte-3 branch that I got working with Symfony 5, and it looks like you've fix the autowiring issues and such, so that AdminLTE bundle should be okay. BUT FOSUserBundle no longer works with Symfony 5, so updating the demo repo requires proper configuration.  

Can you provide some guidance?  In particular, can you rename the adminlte-3 branch to something like bootstrap-4 (and merge master into it so we get all the new service goodies)?  I'd like to try to update the demo bundle with Symfony 5, but I really don't want to do it with bootstrap 3, which is feeling pretty outdated.

I've been playing around with a few of the bootstrap 5 admin themes, I'm pretty jazzed about dropping jQuery for everything except DataTables.

Thanks,

Tac

**Comment by kevinpapst** [comment]
In my own App I am currently removing FOSUserBundle and starting to reduce the coupling here as well, see #163 for example.

I am not sure about the future of this bundle (regarding the v3 version of AdminLTE). 
For now I am just trying to keep compatibility with newer Symfony and PHP versions. 

If you, or someone else, wants to work on the v3 integration, please help with testing and raising PRs.
For me, there is no rush with that: I don't trust ColorLib with AdminLTE v3 - since they took over AdminLTE there is a very weird release and development cycle. And I am not sure if I even want to upgrade to it.

**Comment by creiner** [comment]
Thanks for your honest opinion regarding the future of this bundle/AdminLTE.
What are the plans for your own projects regarding an admin interface?

**Comment by kevinpapst** [comment]
Using AdminLTE v2, it works perfectly fine and is stable.
There is no need to upgrade to the latest lib versions only for sake of upgrading.

I don't mind working on v3 with the community. But I will not work on it without help.

**Comment by wucherpfennig** [comment]
We have built our main intranet service with the v3 branch 😀 so far no issues.

**Comment by wucherpfennig** [comment]
Out off curiosity: version 3.1 has been released. Do you think you could update the branch to use the new release?

**Comment by kevinpapst** [comment]
> Do you think you could update the branch to use the new release

Do you think you can provide the PR for that @wucherpfennig ?

**Comment by cmmata** [comment]
In a side project I'm working on, I will soon add an intranet, and I was thinking about using this bundle as I was using it in my previous job two years ago. I can help debugging the v3 branch if needed, and also probably contribute with small fixes if I can. But take into account that my frontend knowledge is really outdated.

**Comment by kevinpapst** [comment]
Awesome @cmmata , someone must start and than I will eventually hop back on the bus as well^^

Contributions are required this time to push the branch to a production ready level.

**Comment by cmmata** [comment]
> Awesome @cmmata , someone must start and than I will eventually hop back on the bus as well^^
> 
> Contributions are required this time to push the branch to a production ready level.

Okay, I'll let you know when I start with this. I expect to be in a week or two.

**Comment by wucherpfennig** [comment]
> > Do you think you could update the branch to use the new release
> 
> Do you think you can provide the PR for that @wucherpfennig ?

Honestly I am quite busy solving our intranet issues here 😀 but I am happy to test and provide some feedback

**Comment by kevinpapst** [comment]
@cavasinf as you are using the v3 branch I invite you to this thread.

What I wanted to discuss: is there anyone willing to take over the maintainer ship for v3? 
We could create a new repository for it and link from here. Or we release it under a new major version?

**Comment by cavasinf** [comment]
I'm currently creating a new project which uses AdminLTE 3, 
I've been using V2 since two+ years (and I really appreciate all the hard work :+1:).

> is there anyone willing to take over the maintainer ship for v3?

Because I'm using this bundle for my Clients projects, I cannot be a "full" maintainer.
But what I can do is test, debug and fix all errors that I encounter.

> We could create a new repository for it and link from here. Or we release it under a new major version?

New Major version is a good option IMO. 
If this new Major version is stable and "unmistakable", 
it will be nice if new devs will get the last version 3 "without knowing".
If they want the version 2, they can simply force that major version.

**Comment by kevinpapst** [comment]
Yeah. I will not use AdminLTE v3 and move away from v2 in the future myself. So I am looking for people who want to contribute here.
My own projects will be switched to tabler.io - there is a brand new (but undocumented) bundle as well: https://github.com/kevinpapst/TablerBundle

**Comment by wucherpfennig** [comment]
As I am using a wild mix between branch v3 and bootstrap and adminlteV3 myself...

How hard is the migration to your new bundle? @kevinpapst

**Comment by kevinpapst** [comment]
Tabler uses last version of Bootstrap /v5 instead of v4 as AdminLTE v3), so this alone is work.

The bundle is based on this bundle here and I tried to keep compatibility, but at some point there were necessary changes (especially embed/include names).

So definitely not a simple package switch, which I guess you didn't expect.
It is possible and it is worth the effort, as Tabler is a great framework. But it needs some work... how much depends on the size of your app and the amount of twig files.

**Comment by cavasinf** [comment]
> As I am using a wild mix between branch v3 and bootstrap and adminlteV3 myself...
@wucherpfennig  What is the reason to that ? (just curious)

**Comment by cavasinf** [comment]
@kevinpapst I'm looking towards your new `tabler` bundle.
I know it rough to ask that, but can you provide any "current state" / WIP / "Roadmap" in your README bundle.

When most of my "needs" will be done, it will be pleasure to switch to your new bundle.

**Comment by kevinpapst** [comment]
The Tabler bundle is pretty stable. I migrated my time-tracking application to tabler (not yet released) and all I needed is in the bundle. So I wouldn't expect big changes, "only" the docs and demo-application are missing.

**Comment by tacman** [comment]
I'd be interested in helping out on a demo app.  Can you provide a shell
with a working menu?

What do you think of adding KnpMenuBundle support to TablerBundle?  All of
my projects use it, it'd make things much easier for me to help with a demo
if that were working.

For the demo app, how about recreating (one step at a time) the tabler demo
at https://preview.tabler.io/, but in PHP using the bundle?  We could start
with the menu links all going to a "Coming Soon" page, and then selectively
add the pages.

Tac

**Comment by cavasinf** [comment]
Just coming out from brainstorming, 
we will use your (@kevinpapst) new tabler bundle.

No need for us to learn how to manipulate AdminLTE 3 if you are developing the Symfony bundle for Tabler (which is more nicer than Admin 3).

Due to that, I'll no longer implementing and using the V3 branch.
Instead of that, I'll provide you any feedback/PR for your Tabler bundle if I stumble into anything.

Starting the project next Monday. 🎉

**Comment by JaumeAG** [comment]
How can I update my adminlte version? and is it stable?
